account_payment_term_extension/models/account_payment_term.py

```diff
@@ -187,7 +202,15 @@ def compute(self, value, date_ref=False, currency=None):
         precision_digits = currency.decimal_places
         next_date = fields.Date.from_string(date_ref)
         for line in self.line_ids:
-            amt = line.compute_line_amount(value, amount, precision_digits)
+            if line.value == "amount_untaxed" and last_stock_move:
+                amount2 = amount
+                if amount2 == value:
```

Comment:
Why do you need this comparison?

Reply:
Removed now. Really it's not necessary according to tests.